pkg/controller/jobframework/reconciler.go

```diff
@@ -357,9 +357,9 @@ func (r *JobReconciler) ReconcileGenericJob(ctx context.Context, req ctrl.Reques
 
 	// 1. make sure there is only a single existing instance of the workload.
 	// If there's no workload exists and job is unsuspended, we'll stop it immediately.
-	wl, err := r.ensureOneWorkload(ctx, job, object)
-	if err != nil {
-		return ctrl.Result{}, err
+	wl, requeue, err := r.ensureOneWorkload(ctx, job, object)
```

Comment:
Why do we need the requeue? If there is an error then "requeue" is implied. Also, I suppose whenever a new pod is created it triggers the reconciler so, not sure why we need to force the requeue. It would also make the diff smaller.

Reply:
You right. We can avoid requeue. Reverted. Thanks!